order-book/src/test/java/com/market/banica/order/book/service/grpc/componentTests/OrderBookComponentIT.java

```diff
@@ -83,31 +96,181 @@
+                .enableRetry().maxRetryAttempts(MAX_RETRY_ATTEMPTS).directExecutor().build();
 
         asynchronousStub = AuroraServiceGrpc.newStub(channelTwo);
+        auroraBlockingStub = AuroraServiceGrpc.newBlockingStub(channelThree);
         blockingStub = OrderBookServiceGrpc.newBlockingStub(grpcCleanup.register(channel));
     }
 
+    @Test
+    public void orderBookServiceToMarketThroughAuroraRetryExecutesWithSuccess() throws IOException {
+        //Arrange
+        int numberOfTickResponses = 10;
+        ReflectionTestUtils.setField(orderBookService, AURORA_CLIENT, auroraClient);
+
+        String serverNameMarket = InProcessServerBuilder.generateName();
+
+        when(auroraClient.getAsynchronousStub()).thenReturn(asynchronousStub);
+
+        ManagedChannel marketChannel = InProcessChannelBuilder.forName(serverNameMarket)
+                .defaultServiceConfig(ChannelRPCConfig.getInstance().getServiceConfig())
+                .enableRetry().maxRetryAttempts(MAX_RETRY_ATTEMPTS).directExecutor().build();
+
+        // Start Market
+        startMarketService(numberOfTickResponses, serverNameMarket, THREAD_SLEEP_TIME_MARKET);
+
+        // Start OrderBook
+        startOrderBookService(orderBookService, THREAD_SLEEP_TIME_DEFAULT);
+
+        List<TickResponse> tickResponses = new ArrayList<>();
+
+        // Start Aurora
+        startAuroraService(marketChannel, tickResponses);
+
+        InterestsRequest interestsRequest = InterestsRequest.newBuilder().setClientId(clientId).setItemName(EGGS_GOOD).build();
+
+
+        //Act
+        InterestsResponse interestsResponse = blockingStub.announceItemInterest(interestsRequest);
+
+        //Assert
+        assertTrue(interestsResponse.isInitialized());
+        assertEquals(numberOfTickResponses, tickResponses.size());
+    }
+
+    @Test
+    public void auroraServiceToOrderBookRequestsRetryExecutesWithSuccess() {
+        //Arrange
+        ItemOrderBookResponse response = generateItemOrderBookExpectedResponse();
+
+        new Thread(() -> {
+            try {
+                Thread.sleep(THREAD_SLEEP_TIME_DEFAULT);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverName).directExecutor().addService(testConfiguration.getGrpcOrderBookServiceItemLayers(response))
+                        .build().start());
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        }).start();
+
+        ItemOrderBookRequest auroraRequest = ItemOrderBookRequest.newBuilder()
+                .setItemName(productName).setClientId(clientId).setQuantity(productQuantity).build();
+
+        //Act
+        ItemOrderBookResponse auroraResponse = blockingStub.getOrderBookItemLayers(auroraRequest);
+
+        ItemOrderBookResponse expected = generateItemOrderBookExpectedResponse();
```

Comment:
expected should be part of Arrange, not Act

Reply:
Fixed